config/routes.rb

```diff
@@ -5,4 +5,6 @@
   end
   resources :categories, only: [:show]
   devise_for :users
+  resources :subscriptions, only: [:new]
+  post '/charges', controller: :subscriptions, action: :create
```

Comment:
try to use restful routes, add `:create` to resources and update the url in the view.